[Automated Report] com.intellij.diagnostic.PluginException: Invalid PSI Element: class net.seesharpsoft.intellij.plugins.csv.psi.CsvFile #csv  because: file:///Users/jaapbranderhorst/IdeaProjects/SBWZ_Docs-datamodel/docs/designs/boekhouding-tasks.tsv (invalid) is invalid

Message
---
<no message>

Stacktrace
---
com.intellij.diagnostic.PluginException: Invalid PSI Element: class net.seesharpsoft.intellij.plugins.csv.psi.CsvFile #csv  because: file:///Users/jaapbranderhorst/IdeaProjects/SBWZ_Docs-datamodel/docs/designs/boekhouding-tasks.tsv (invalid) is invalid
invalidated at: see attachment [Plugin: net.seesharpsoft.intellij.plugins.csv]
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:23)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:100)
	at com.intellij.psi.util.PsiUtilCore.ensureValid(PsiUtilCore.java:509)
	at com.intellij.psi.impl.source.PsiFileImpl.getTextLength(PsiFileImpl.java:349)
	at com.intellij.psi.impl.source.PsiFileImpl.getTextRange(PsiFileImpl.java:355)
	at com.intellij.codeInsight.daemon.impl.FileStatusMap.getFileDirtyScope(FileStatusMap.java:224)
	at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.isErrorAnalyzingFinished(DaemonCodeAnalyzerImpl.java:931)
	at com.anthropic.code.plugin.mcp.tools.DiagnosticTools$addTools$1$1$1.daemonFinished(DiagnosticTools.kt:106)
	at com.intellij.codeInsight.daemon.DaemonCodeAnalyzer$DaemonListener.daemonFinished(DaemonCodeAnalyzer.java:123)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeMethod(MessageBusImpl.kt:820)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:764)
	at com.intellij.util.messages.impl.MessageBusImplKt.executeOrAddToQueue(MessageBusImpl.kt:585)
	at com.intellij.util.messages.impl.MessagePublisher.publish$intellij_platform_core(MessageBusImpl.kt:556)
	at com.intellij.util.messages.impl.MessagePublisher.invoke(MessageBusImpl.kt:533)
	at jdk.proxy2/jdk.proxy2.$Proxy85.daemonFinished(Unknown Source)
	at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl$MyDaemonProgressIndicator.onStop(DaemonCodeAnalyzerImpl.java:1685)
	at com.intellij.codeInsight.daemon.impl.DaemonProgressIndicator.stop(DaemonProgressIndicator.java:39)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService.lambda$applyInformationToEditorsLater$3(PassExecutorService.java:554)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:239)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:25)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:201)
	at com.intellij.openapi.application.impl.AppImplKt$runnableUnitFunction$1.invoke(appImpl.kt:124)
	at com.intellij.openapi.application.impl.AppImplKt$runnableUnitFunction$1.invoke(appImpl.kt:124)
	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.doRunWriteIntentReadAction(NestedLocksThreadingSupport.kt:736)
	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.runPreventiveWriteIntentReadAction(NestedLocksThreadingSupport.kt:710)
	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.runWriteIntentReadAction(NestedLocksThreadingSupport.kt:664)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:1048)
	at com.intellij.openapi.application.impl.ApplicationImpl$7.lambda$run$0(ApplicationImpl.java:602)
	at com.intellij.concurrency.ThreadContext.installThreadContext(threadContext.kt:305)
	at com.intellij.openapi.application.impl.ApplicationImpl$7.run(ApplicationImpl.java:600)
	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:167)
	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:167)
	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:173)
	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:167)
	at com.intellij.util.concurrency.ContextRunnable.lambda$run$0(ContextRunnable.java:26)
	at com.intellij.concurrency.ThreadContext.resetThreadContext(threadContext.kt:294)
	at com.intellij.util.concurrency.ContextRunnable.run(ContextRunnable.java:25)
	at com.intellij.openapi.application.impl.NonBlockingFlushQueue.runNextEvent$lambda$1$1$0(NonBlockingFlushQueue.kt:334)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:901)
	at com.intellij.openapi.application.impl.NonBlockingFlushQueue.runNextEvent$lambda$1$1(NonBlockingFlushQueue.kt:333)
	at com.intellij.concurrency.ThreadContext.resetThreadContext(threadContext.kt:294)
	at com.intellij.openapi.application.impl.NonBlockingFlushQueue.runNextEvent$lambda$1(NonBlockingFlushQueue.kt:330)
	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.tryRunWriteIntentReadAction(NestedLocksThreadingSupport.kt:689)
	at com.intellij.openapi.application.impl.NonBlockingFlushQueue.runNextEvent(NonBlockingFlushQueue.kt:326)
	at com.intellij.openapi.application.impl.NonBlockingFlushQueue.flushNow(NonBlockingFlushQueue.kt:305)
	at com.intellij.openapi.application.impl.NonBlockingFlushQueue.FLUSH_NOW$lambda$0(NonBlockingFlushQueue.kt:167)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:728)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:750)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:664)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:517)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$0$0$0(IdeEventQueue.kt:333)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:1065)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:109)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:1065)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$0(IdeEventQueue.kt:331)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:371)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: com.intellij.psi.PsiInvalidElementAccessException: Invalid PSI Element: class net.seesharpsoft.intellij.plugins.csv.psi.CsvFile #csv  because: file:///Users/jaapbranderhorst/IdeaProjects/SBWZ_Docs-datamodel/docs/designs/boekhouding-tasks.tsv (invalid) is invalid
invalidated at: see attachment
	at com.intellij.psi.util.PsiUtilCore.ensureValid(PsiUtilCore.java:508)
	... 64 more


Plugin
---
PluginClassLoader(plugin=PluginMainDescriptor(name=CSV Editor, id=net.seesharpsoft.intellij.plugins.csv, version=4.2.0, isBundled=false, path=~/Library/Application Support/JetBrains/IntelliJIdea2025.3/plugins/intellij-csv-validator), packagePrefix=null, state=active, parents=ContentModuleDescriptor(id=intellij.platform.vcs.log.graph) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.relaxng) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.vcs.log) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.vcs.dvcs) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.scriptDebugger.protocolReaderRuntime) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.scriptDebugger.backend) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.scriptDebugger.ui) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.tasks) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.collaborationTools.auth.base) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.collaborationTools.auth) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.vcs.impl) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.vcs.log.impl) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.vcs.dvcs.impl) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.collaborationTools) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.spellchecker) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.spellchecker.xml) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.platform.tasks.impl) <- PluginMainDescriptor(name=IDEA CORE, id=com.intellij, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/lib), ContentModuleDescriptor(id=intellij.vcs.github, loadingRule=EMBEDDED) <- PluginMainDescriptor(name=GitHub, id=org.jetbrains.plugins.github, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/plugins/vcs-github), ContentModuleDescriptor(id=intellij.vcs.github.ultimate.features) <- PluginMainDescriptor(name=GitHub, id=org.jetbrains.plugins.github, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/plugins/vcs-github), ContentModuleDescriptor(id=intellij.vcs.github.ultimate.cron) <- PluginMainDescriptor(name=GitHub, id=org.jetbrains.plugins.github, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/plugins/vcs-github), ContentModuleDescriptor(id=intellij.vcs.github.ultimate.docker) <- PluginMainDescriptor(name=GitHub, id=org.jetbrains.plugins.github, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/plugins/vcs-github), ContentModuleDescriptor(id=intellij.vcs.github.json) <- PluginMainDescriptor(name=GitHub, id=org.jetbrains.plugins.github, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/plugins/vcs-github), ContentModuleDescriptor(id=intellij.vcs.github.tracker) <- PluginMainDescriptor(name=GitHub, id=org.jetbrains.plugins.github, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/plugins/vcs-github), ContentModuleDescriptor(id=intellij.vcs.github.ultimatePromo) <- PluginMainDescriptor(name=GitHub, id=org.jetbrains.plugins.github, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/plugins/vcs-github), ContentModuleDescriptor(id=intellij.vcs.github.git) <- PluginMainDescriptor(name=GitHub, id=org.jetbrains.plugins.github, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/plugins/vcs-github), PluginMainDescriptor(name=GitHub, id=org.jetbrains.plugins.github, version=253.30387.90, isBundled=true, path=/Applications/IntelliJ IDEA.app/Contents/plugins/vcs-github), )

IDE
---
IntelliJ IDEA (IU-253.30387.90)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,23 +10,19 @@
 
 ### Fixed
 
-## 4.2.3 - Jun 21, 2026
-
-### Fixed
-
-- branch code for retrieving plugin info for legacy vs EAP version
-
-## 4.2.2 - Jun 21, 2026
+## 4.3.0 - Jun 21, 2026
 
 ### Changed
 
 - replaced deprecated function ReadAction.compute(ThrowableComputable)
 - replaced internal function PluginManagerCore.getPlugin(PluginId)
-
-## 4.2.1 - May 24, 2026
+- branch code for retrieving plugin info for legacy vs EAP version
 
 ### Fixed
 
+- Fixed `NoClassDefFoundError` in `CsvEditorSettings`
+- Fixed `java.lang.Throwable: ... Class initialization must not depend on services` by removing service access from class initializers and constructors
+- Fixed a `PluginException` (caused by `java.lang.IllegalArgumentException`) that occurred during early plugin initialization
 - PluginException: Invalid PSI Element CsvFile when file is invalidated during annotation, intention actions, or inspection fixes #964
 - IncorrectOperationException: parent CsvTableEditorSwing already disposed when async PsiTreeChangeListener registration completes #962
 - Modified `CsvPlugin.openLink` to use `executeOnPooledThread` for setting dialogs #953

--- a/gradle.properties
+++ b/gradle.properties
@@ -4,7 +4,7 @@
 
 pluginName=CSV Editor
 pluginId=net.seesharpsoft.intellij.plugins.csv
-pluginVersion=4.2.3
+pluginVersion=4.3.0
 
 pluginSinceBuild=242
 

--- a/src/main/java/net/seesharpsoft/intellij/plugins/csv/CsvPluginDescriptorRetriever.java
+++ b/src/main/java/net/seesharpsoft/intellij/plugins/csv/CsvPluginDescriptorRetriever.java
@@ -14,9 +14,15 @@ public final class CsvPluginDescriptorRetriever {
     private static final PluginId PLUGIN_ID = PluginId.getId("net.seesharpsoft.intellij.plugins.csv");
 
     public static IdeaPluginDescriptor getPluginDescriptor() {
-        BuildNumber buildNumber = ApplicationInfo.getInstance().getBuild();
+        BuildNumber buildNumber = null;
+        try {
+            buildNumber = ApplicationInfo.getInstance().getBuild();
+        } catch (Exception e) {
+            LOG.debug("Failed to retrieve build number", e);
+        }
+
         // PluginDetailsService is available from 2026.2, which roughly corresponds to build 262
-        if (buildNumber.getBaselineVersion() >= 262) {
+        if (buildNumber != null && buildNumber.getBaselineVersion() >= 262) {
             try {
                 Class<?> serviceClass = Class.forName("com.intellij.ide.plugins.PluginDetailsService");
                 Method getInstanceMethod = serviceClass.getMethod("getInstance");

--- a/src/main/java/net/seesharpsoft/intellij/plugins/csv/CsvPluginManager.java
+++ b/src/main/java/net/seesharpsoft/intellij/plugins/csv/CsvPluginManager.java
@@ -11,7 +11,12 @@ public final class CsvPluginManager {
 
     public static ResourceBundle getResourceBundle() {
         if (_resourceBundle == null) {
-            _resourceBundle = DynamicBundle.getPluginBundle(getPluginDescriptor());
+            IdeaPluginDescriptor descriptor = getPluginDescriptor();
+            if (descriptor != null) {
+                _resourceBundle = DynamicBundle.getPluginBundle(descriptor);
+            } else {
+                return ResourceBundle.getBundle("localization.CsvEditorResources");
+            }
         }
         return _resourceBundle;
     }

--- a/src/main/java/net/seesharpsoft/intellij/plugins/csv/highlighter/CsvTextAttributeKeys.java
+++ b/src/main/java/net/seesharpsoft/intellij/plugins/csv/highlighter/CsvTextAttributeKeys.java
@@ -34,26 +34,36 @@ public final class CsvTextAttributeKeys {
             createTextAttributesKey("CSV_BAD_CHARACTER", HighlighterColors.BAD_CHARACTER);
 
     public static final Integer MAX_COLUMN_COLORING_COLORS = 10;
-    public static final AttributesDescriptor[] DESCRIPTORS;
+    private static AttributesDescriptor[] DESCRIPTORS = null;
+
+    public static AttributesDescriptor[] getDescriptors() {
+        if (DESCRIPTORS == null) {
+            List<AttributesDescriptor> attributesDescriptors = new ArrayList<>();
+            ResourceBundle bundle = getResourceBundle();
+            attributesDescriptors.add(new AttributesDescriptor(bundle.getString("color.attribute.separator"), CsvTextAttributeKeys.COMMA));
+            attributesDescriptors.add(new AttributesDescriptor(bundle.getString("color.attribute.quote"), CsvTextAttributeKeys.QUOTE));
+            attributesDescriptors.add(new AttributesDescriptor(bundle.getString("color.attribute.text"), CsvTextAttributeKeys.TEXT));
+            attributesDescriptors.add(new AttributesDescriptor(bundle.getString("color.attribute.text.escaped"), CsvTextAttributeKeys.ESCAPED_TEXT));
+            attributesDescriptors.add(new AttributesDescriptor(bundle.getString("color.attribute.comment"), CsvTextAttributeKeys.COMMENT));
+
+            for (int i = 0; i < MAX_COLUMN_COLORING_COLORS; ++i) {
+                TextAttributesKey textAttributesKey = COLUMN_COLORING_ATTRIBUTES.get(i);
+                attributesDescriptors.add(new AttributesDescriptor(String.format(bundle.getString("color.attribute.column.nr"), i + 1), textAttributesKey));
+            }
+            DESCRIPTORS = attributesDescriptors.toArray(new AttributesDescriptor[0]);
+        }
+        return DESCRIPTORS;
+    }
+
     private static final List<TextAttributesKey> COLUMN_COLORING_ATTRIBUTES;
     private static final Key<List<TextAttributes>> COLUMN_COLORING_TEXT_ATTRIBUTES = Key.create("CSV_PLUGIN_COLUMN_COLORING_ATTRIBUTES");
 
     static {
-        List<AttributesDescriptor> attributesDescriptors = new ArrayList<>();
-        ResourceBundle bundle = getResourceBundle();
-        attributesDescriptors.add(new AttributesDescriptor(bundle.getString("color.attribute.separator"), CsvTextAttributeKeys.COMMA));
-        attributesDescriptors.add(new AttributesDescriptor(bundle.getString("color.attribute.quote"), CsvTextAttributeKeys.QUOTE));
-        attributesDescriptors.add(new AttributesDescriptor(bundle.getString("color.attribute.text"), CsvTextAttributeKeys.TEXT));
-        attributesDescriptors.add(new AttributesDescriptor(bundle.getString("color.attribute.text.escaped"), CsvTextAttributeKeys.ESCAPED_TEXT));
-        attributesDescriptors.add(new AttributesDescriptor(bundle.getString("color.attribute.comment"), CsvTextAttributeKeys.COMMENT));
-
         COLUMN_COLORING_ATTRIBUTES = new ArrayList<>();
         for (int i = 0; i < MAX_COLUMN_COLORING_COLORS; ++i) {
             TextAttributesKey textAttributesKey = createTextAttributesKey(String.format("CSV_PLUGIN_COLUMN_COLORING_ATTRIBUTE_%d", i), CsvTextAttributeKeys.TEXT);
             COLUMN_COLORING_ATTRIBUTES.add(textAttributesKey);
-            attributesDescriptors.add(new AttributesDescriptor(String.format(bundle.getString("color.attribute.column.nr"), i + 1), textAttributesKey));
         }
-        DESCRIPTORS = attributesDescriptors.toArray(new AttributesDescriptor[0]);
     }
 
     public static TextAttributesKey getTextAttributesKeys(int columnIndex) {

--- a/src/main/java/net/seesharpsoft/intellij/plugins/csv/settings/CsvColorSettings.java
+++ b/src/main/java/net/seesharpsoft/intellij/plugins/csv/settings/CsvColorSettings.java
@@ -48,7 +48,7 @@ public class CsvColorSettings implements ColorSettingsPage {
     @NotNull
     @Override
     public AttributesDescriptor[] getAttributeDescriptors() {
-        return CsvTextAttributeKeys.DESCRIPTORS;
+        return CsvTextAttributeKeys.getDescriptors();
     }
 
     @NotNull

--- a/src/main/java/net/seesharpsoft/intellij/plugins/csv/settings/CsvEditorSettings.java
+++ b/src/main/java/net/seesharpsoft/intellij/plugins/csv/settings/CsvEditorSettings.java
@@ -47,13 +47,17 @@ public class CsvEditorSettings implements PersistentStateComponent<CsvEditorSett
         TEXT_ONLY("settings.editor.prio.text_only");
 
         private final String resourceKey;
+        private String display;
 
         private EditorPrio(String resourceKey) {
             this.resourceKey = resourceKey;
         }
 
         public String getDisplay() {
-            return getLocalizedText(resourceKey);
+            if (display == null) {
+                display = getLocalizedText(resourceKey);
+            }
+            return display;
         }
     }
 
@@ -62,13 +66,17 @@ public class CsvEditorSettings implements PersistentStateComponent<CsvEditorSett
         SIMPLE("settings.editor.coloring.simple");
 
         private final String resourceKey;
+        private String display;
 
         ValueColoring(String resourceKey) {
             this.resourceKey = resourceKey;
         }
 
         public String getDisplay() {
-            return getLocalizedText(this.resourceKey);
+            if (display == null) {
+                display = getLocalizedText(this.resourceKey);
+            }
+            return display;
         }
     }
 

--- a/src/main/java/net/seesharpsoft/intellij/plugins/csv/settings/CsvEditorSettings.java
+++ b/src/main/java/net/seesharpsoft/intellij/plugins/csv/settings/CsvEditorSettings.java
@@ -42,33 +42,33 @@ public class CsvEditorSettings implements PersistentStateComponent<CsvEditorSett
     private static final CsvEditorSettings STATIC_TEST_INSTANCE = new CsvEditorSettings();
 
     public enum EditorPrio {
-        TEXT_FIRST(getLocalizedText("settings.editor.prio.text_first")),
-        TABLE_FIRST(getLocalizedText("settings.editor.prio.table_first")),
-        TEXT_ONLY(getLocalizedText("settings.editor.prio.text_only"));
+        TEXT_FIRST("settings.editor.prio.text_first"),
+        TABLE_FIRST("settings.editor.prio.table_first"),
+        TEXT_ONLY("settings.editor.prio.text_only");
 
-        private final String label;
-        
-        private EditorPrio(String label) {
-            this.label = label;
+        private final String resourceKey;
+
+        private EditorPrio(String resourceKey) {
+            this.resourceKey = resourceKey;
         }
 
         public String getDisplay() {
-            return label;
+            return getLocalizedText(resourceKey);
         }
     }
 
     public enum ValueColoring {
-        RAINBOW(getLocalizedText("settings.editor.coloring.rainbow")),
-        SIMPLE(getLocalizedText("settings.editor.coloring.simple"));
+        RAINBOW("settings.editor.coloring.rainbow"),
+        SIMPLE("settings.editor.coloring.simple");
 
-        private final String display;
+        private final String resourceKey;
 
-        ValueColoring(String displayArg) {
-            this.display = displayArg;
+        ValueColoring(String resourceKey) {
+            this.resourceKey = resourceKey;
         }
 
         public String getDisplay() {
-            return this.display;
+            return getLocalizedText(this.resourceKey);
         }
     }
 

--- a/src/test/java/net/seesharpsoft/intellij/plugins/csv/settings/CsvEditorSettingsTest.java
+++ b/src/test/java/net/seesharpsoft/intellij/plugins/csv/settings/CsvEditorSettingsTest.java
@@ -21,4 +21,13 @@ public class CsvEditorSettingsTest extends CsvBasePlatformTestCase {
 
         assertEquals(CsvEditorSettings.ESCAPE_CHARACTER_DEFAULT, CsvHelper.getEscapeCharacter(myFixture.getFile()));
     }
+
+    public void testGetDisplayDoesNotThrow() {
+        for (CsvEditorSettings.EditorPrio prio : CsvEditorSettings.EditorPrio.values()) {
+            assertNotNull(prio.getDisplay());
+        }
+        for (CsvEditorSettings.ValueColoring coloring : CsvEditorSettings.ValueColoring.values()) {
+            assertNotNull(coloring.getDisplay());
+        }
+    }
 }

